### ISSUE TITLE
Overflow in pcl::octree::OctreeIteratorBase< OctreeT >::getNodeID()

### ISSUE THREAD

**Issue by rhuitl** [PointCloudLibrary/pcl#1296]
The method
```c++
  unsigned long pcl::octree::OctreeIteratorBase< OctreeT >::getNodeID()
```
is supposed to return an unsigned long (64 bit) value, however the computation is carried out with only 32 bit precision. For spatially large point clouds this results in duplicate node IDs.

The fix is simple: cast key.x, key.y and key.z to unsigned long like this:
```c++
  id = (unsigned long)key.x << (depth * 2) | (unsigned long)key.y << (depth * 1) | (unsigned long)key.z << (depth * 0);
```

**Comment by rhuitl** [comment]
Fixed in pull request #1297

**Comment by jspricke** [comment]
merged some time ago.
